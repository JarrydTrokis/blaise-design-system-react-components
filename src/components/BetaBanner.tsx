import React, {ReactElement} from "react";

function BetaBanner(): ReactElement {
    return (
        <>
            <div className="ons-phase-banner">
                <div className="ons-container ">
                    <div className="ons-grid ons-grid--flex ons-grid--gutterless ons-grid--vertical-center ons-grid--no-wrap">
                        <div className="ons-grid__col ons-col-auto ons-u-flex-no-grow">
                            <h3 className="ons-phase-banner__badge">BETA</h3>
                        </div>
                        <div className="ons-grid__col ons-col-auto ons-u-flex-shrink">
                            <p className="ons-phase-banner__desc ons-u-fs-s ons-u-mb-no">This is a new service â€“ your <a
                                href="https://ons.service-now.com/">feedback</a> will help us improve it.</p>
                        </div>
                    </div>
                </div>
            </div>
        </>
    );
}

export default BetaBanner;
