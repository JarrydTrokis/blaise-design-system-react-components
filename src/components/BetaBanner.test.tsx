import React from "react";
import { cleanup, render, screen } from "@testing-library/react";
import BetaBanner from "./BetaBanner";

describe("ONS In Dev Banner Test", () => {
    afterEach(() => {
        cleanup();
    });

    it("matches Snapshot", () => {
        expect(render(<BetaBanner />)).toMatchSnapshot();
    });

    it("should render correctly", () => {
        render(<BetaBanner />);
        expect(screen.getByText(/This is a new service/i)).toBeVisible();
        expect(screen.getByText(/BETA/i)).toBeVisible();
    });
});
