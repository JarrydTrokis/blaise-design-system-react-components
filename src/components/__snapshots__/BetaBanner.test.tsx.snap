// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ONS In Dev Banner Test matches Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ons-phase-banner"
      >
        <div
          class="ons-container "
        >
          <div
            class="ons-grid ons-grid--flex ons-grid--gutterless ons-grid--vertical-center ons-grid--no-wrap"
          >
            <div
              class="ons-grid__col ons-col-auto ons-u-flex-no-grow"
            >
              <h3
                class="ons-phase-banner__badge"
              >
                BETA
              </h3>
            </div>
            <div
              class="ons-grid__col ons-col-auto ons-u-flex-shrink"
            >
              <p
                class="ons-phase-banner__desc ons-u-fs-s ons-u-mb-no"
              >
                This is a new service – your 
                <a
                  href="https://ons.service-now.com/"
                >
                  feedback
                </a>
                 will help us improve it.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ons-phase-banner"
    >
      <div
        class="ons-container "
      >
        <div
          class="ons-grid ons-grid--flex ons-grid--gutterless ons-grid--vertical-center ons-grid--no-wrap"
        >
          <div
            class="ons-grid__col ons-col-auto ons-u-flex-no-grow"
          >
            <h3
              class="ons-phase-banner__badge"
            >
              BETA
            </h3>
          </div>
          <div
            class="ons-grid__col ons-col-auto ons-u-flex-shrink"
          >
            <p
              class="ons-phase-banner__desc ons-u-fs-s ons-u-mb-no"
            >
              This is a new service – your 
              <a
                href="https://ons.service-now.com/"
              >
                feedback
              </a>
               will help us improve it.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
